nxcomp/Log.cpp

```diff
@@ -0,0 +1,81 @@
+
+#include <stdio.h>
+#include <iostream>
+#include <fstream>
+#include <sstream>
+
+
+#include "Log.h"
+
+NXLog nx_log;
+
+
+NXLog::NXLog(char *filename)
+{
+	std::fstream *file = new std::fstream();
+	file->open(filename);
+	stream_ = file;
+	level_ = DEBUG;
+}
+
+bool NXLog::will_log() const
+{
+	std::map<std::string, NXLogLevel>::const_iterator item = per_file_levels_.find(current_file());
+
+	if ( item != per_file_levels_.end() )
+	{
+		return current_level_ >= item->second;
+	}
+	else
+	{
+		return current_level_ >= level_;
+	}
+}
+
+
+std::string NXLogStamp::to_string() const
+{
+	static const char* level_names[] = {
+		"DBG",
+		"INF",
+		"WRN",
+		"ERR",
+		"FAT"
+	};
+
+	std::ostringstream oss;
+	struct tm *timeinfo;
+	char buffer[32];
+	char buffer2[32];
+
+	oss << ((level_ >=0 && level_ < NXLOG_LEVEL_COUNT ) ? level_names[level_] : "???");
+
+
+	timeinfo = localtime(&timestamp_.tv_sec);
+
+	if (strftime(buffer, sizeof(buffer)-1, "%Y/%m/%d %H:%M:%S", timeinfo) == 0)
```

Comment:
While `strftime` [is supposed to be thread-safe](http://pubs.opengroup.org/onlinepubs/009695399/functions/xsh_chap02_09.html), but `glibc` is [known](https://bugzilla.redhat.com/show_bug.cgi?id=544022) to be broken [beyond repair](https://sourceware.org/bugzilla/show_bug.cgi?id=4350) and we probably should not be using it. Sadly.